mpqp/execution/connection/aws_connection.py

```diff
@@ -49,48 +72,197 @@ def setup_aws_braket_account() -> tuple[str, list[Any]]:
+    """
+
+    credentials_file = Path.home() / ".aws" / "credentials"
+
+    credentials_dir = credentials_file.parent
+    if not credentials_dir.exists():
+        credentials_dir.mkdir(parents=True, exist_ok=True)
+
+    config = ConfigParser()
+    if credentials_file.exists():
+        config.read(credentials_file)
+
+    if config.has_section(profile_name):
+        config.remove_section(profile_name)
+
+    config.add_section(profile_name)
+    config[profile_name]["aws_access_key_id"] = access_key_id
+    config[profile_name]["aws_secret_access_key"] = secret_access_key
+
+    if session_token:
+        config[profile_name]["aws_session_token"] = session_token
+
+    config[profile_name]["region"] = region
+
+    with open(credentials_file, "w") as f:
+        config.write(f)
+
+
+def configure_account_iam() -> tuple[str, list[Any]]:
+    """Configure IAM authentication for Amazon Braket."""
+
+    print("Configuring IAM authentication for Amazon Braket...")
+    os.system("aws configure")
+
+    print("IAM authentication configured successfully.")
+    save_env_variable("BRAKET_AUTH_METHOD", "IAM")
+    save_env_variable("BRAKET_CONFIGURED", "True")
+
+    credentials_file = Path.home() / ".aws" / "credentials"
+
+    config = ConfigParser()
+    config.read(credentials_file)
+
+    access_key_id = config.get("default", "aws_access_key_id", fallback="")
+    secret_access_key = config.get("default", "aws_secret_access_key", fallback="")
+    region = config.get("default", "region", fallback="us-east-1")
+
+    update_aws_credentials_file(
+        profile_name="default",
+        access_key_id=access_key_id,
+        secret_access_key=secret_access_key,
+        session_token=None,
+        region=region,
+    )
+
+    return "IAM configuration successful.", []
+
+
+def get_user_sso_credentials() -> Union[dict[str, str], None]:
+
+    print("Please enter your AWS SSO credentials:")
+
+    try:
+        access_key_id = input("Enter AWS access key ID: ").strip()
+        secret_access_key = getpass("Enter AWS secret access key (hidden): ").strip()
+        session_token = getpass("Enter SSO session token (hidden): ").strip()
+        region = input("Enter SSO region: ").strip()
+
+        return {
+            "access_key_id": access_key_id,
+            "secret_access_key": secret_access_key,
+            "session_token": session_token,
+            "region": region,
+        }
+
+    except Exception as e:
+        print(f"An error occurred while getting credentials: {e}")
+        return None
+
+
+def configure_account_sso() -> tuple[str, list[Any]]:
+    """Configure SSO authentication for Amazon Braket.
+    This function guides the user through the Amazon Braket SSO configuration process.
+    """
+    print("Configuring SSO authentication for Amazon Braket...")
+    sso_credentials = get_user_sso_credentials()
+
+    print("SSO authentication configured successfully.")
```

Comment:
move this after the raising of exception ?

Reply:
Yes indeed this is a mistake ``print("SSO authentication configured successfully.")`` before the exception!!